src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "../Common/Utils.h"
+
+namespace CalculatorApp::ViewModel
+{
+#pragma once
+    [Windows::UI::Xaml::Data::Bindable] public ref class GraphingSettingsViewModel sealed : public Windows::UI::Xaml::Data::INotifyPropertyChanged
+    {
+    public:
+        OBSERVABLE_OBJECT();
+        OBSERVABLE_PROPERTY_R(bool, YMinError);
+        OBSERVABLE_PROPERTY_R(bool, XMinError);
+        OBSERVABLE_PROPERTY_R(bool, XMaxError);
+        OBSERVABLE_PROPERTY_R(bool, YMaxError);
+        OBSERVABLE_PROPERTY_R(GraphControl::Grapher ^, Graph);
+
+        GraphingSettingsViewModel();
+
+        property bool XError
+        {
+            bool get()
+            {
+                return !m_XMinError && !m_XMaxError && m_XMinValue >= m_XMaxValue;
+            }
+        }
+
+        property bool YError
+        {
+            bool get()
+            {
+                return !m_YMinError && !m_YMaxError && m_YMinValue >= m_YMaxValue;
+            }
+        }
+
+        property Platform::String ^ XMin
+        {
+            Platform::String ^ get()
+            {
+                return m_XMin;
+            }
+            void set(Platform::String ^ value)
+            {
+                if (m_XMin == value)
+                {
+                    return;
+                }
+                m_XMin = value;
+                if (m_Graph != nullptr)
+                {
+                    try
+                    {
+                        size_t sz;
+                        auto number = std::stod(value->Data(), &sz);
+                        if (value->Length() == sz)
+                        {
+                            m_Graph->XAxisMin = m_XMinValue = number;
+                            XMinError = false;
+                        }
+                        else
+                        {
+                            XMinError = true;
+                        }
+                    }
+                    catch (...)
+                    {
+                        XMinError = true;
+                    }
+                }
+                RaisePropertyChanged("XError");
+                RaisePropertyChanged("XMin");
+                UpdateDisplayRange();
+            }
+        }
+
+        property Platform::String ^  XMax
+        {
+            Platform::String ^ get()
+            {
+                return m_XMax;
+            }
+            void set(Platform::String ^ value)
+            {
+                if (m_XMax == value)
+                {
+                    return;
+                }
+                m_XMax = value;
+                if (m_Graph != nullptr)
+                {
+                    try
+                    {
+                        size_t sz;
+                        auto number = std::stod(value->Data(), &sz);
```

Comment:
If I zoom out far enough, and click the graph settings, the numbers are red and I can never close the flyout